src/Tasks/Microsoft.NET.Build.Tasks/CheckForDuplicateItemMetadata.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.Build.Framework;
+using System.Linq;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public class CheckForDuplicateItemMetadata : TaskBase
+    {
+        [Required]
+        public ITaskItem[] Items { get; set; }
+
+        [Required]
+        public string MetadataName { get; set; }
+
+        [Output]
+        public ITaskItem[] DuplicateItems { get; set; }
+
+        [Output]
+        public string[] DuplicatedMetadata { get; set; }
+
+        [Output]
+        public bool DuplicatesExist { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            var groupings = Items.GroupBy(item => item.GetMetadata(MetadataName));
+            DuplicatesExist = groupings.Where(g => g.Count() > 1).Any();
```

Comment:
`GroupBy` is lazy, so this method is doing the grouping (and filtering) three times.  I'd do this to avoid this:

```c#
var groupsWithDuplicates = Items.GroupBy(item => item.GetMetadata(MetadataName))
                                .Where(g => g.Count() > 1)
                                .ToList();
```